### PR TITLE
Feature/psa

### PR DESCRIPTION
**Pull request by MathiasOhrstrom**
Item from design: [home_tab.vma]

### Screenshots

Screenshot 1
image
Screenshot 2
image

...

### Checklist

- [x] Functionality is covered by an integration test
- [x] Functionality is accessible in engineering mode

### CODE REVIEW

File: app/src/main/java/com/hedvig/app/feature/home/ui/HomeItemDecoration.kt [hunk 1]
```diff
@@ -46,6 +53,31 @@ class HomeItemDecoration : RecyclerView.ItemDecoration() {
...
+    }
+
+    override fun onDraw(c: Canvas, parent: RecyclerView, state: RecyclerView.State) {
+        val adapter = (parent.adapter as? HomeAdapter)
+        parent.children.forEach { view ->
+            val position = parent.getChildAdapterPosition(view)
+            val item = adapter?.items?.getOrNull(position) ?: return
+
+            if (item is HomeModel.PSA) {
+                divider?.draw(c)
```
Comment (hedvigoscar) [inline app/src/main/java/com/hedvig/app/feature/home/ui/HomeItemDecoration.kt:78]: If every PSA-item simply has a divider, can it not be added directly into its layout? That would simplify our code quite a bit 🙂

File: app/src/main/java/com/hedvig/app/feature/home/ui/HomeModel.kt [hunk 2]
```diff
@@ -35,14 +35,11 @@ sealed class HomeModel {
 
     object StartClaimContained : HomeModel()
 
-    data class HowClaimsWork(val pages: List<HomeQuery.HowClaimsWork>): HomeModel()
+    object InfoCard : HomeModel()
```
Comment (hedvigoscar) [inline app/src/main/java/com/hedvig/app/feature/home/ui/HomeModel.kt:38]: Is this now only a `ConnectPayin`? If so, we should rename it 🙂

File: app/src/main/java/com/hedvig/app/ui/decoration/CenterItemDecoration.kt [hunk 3]
```diff
@@ -2,6 +2,7 @@ package com.hedvig.app.ui.decoration
 
 import android.graphics.Rect
 import android.view.View
+import androidx.core.content.ContextCompat
```
Comment (hedvigoscar) [inline app/src/main/java/com/hedvig/app/ui/decoration/CenterItemDecoration.kt:5]:
```suggestion
```

File: app/src/main/java/com/hedvig/app/ui/decoration/ItemDividerDecoration.kt [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+import android.graphics.Canvas
+import android.graphics.Rect
+import android.graphics.drawable.Drawable
+import android.view.View
+import androidx.core.content.ContextCompat
+import androidx.core.view.children
+import androidx.recyclerview.widget.RecyclerView
+import com.hedvig.app.R
+
+class ItemDividerDecoration(context: Context) : RecyclerView.ItemDecoration() {
```
Comment (hedvigoscar) [inline app/src/main/java/com/hedvig/app/ui/decoration/ItemDividerDecoration.kt:13]: Is this used?

**Review by hedvigoscar — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/hedvig/app/feature/home/ui/HomeItemDecoration.kt (hunk 1 above)
Comment (MathiasOhrstrom) [reply app/src/main/java/com/hedvig/app/feature/home/ui/HomeItemDecoration.kt:78]: I forgot to add `if (prev is HomeModel.PSA && item is HomeModel.PSA)` so it only to be drawn when there are multiple PSAs

Comment (hedvigoscar) [reply app/src/main/java/com/hedvig/app/feature/home/ui/HomeItemDecoration.kt:78]: Sensible!

File: app/src/main/java/com/hedvig/app/ui/decoration/ItemDividerDecoration.kt (hunk 4 above)
Comment (MathiasOhrstrom) [reply app/src/main/java/com/hedvig/app/ui/decoration/ItemDividerDecoration.kt:13]: Nope 🗑️

File: app/src/androidTest/java/com/hedvig/app/feature/home/ActiveWithMultiplePSAsAndConnectPayment.kt [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+        },
+        HomeQuery.QUERY_DOCUMENT to apolloResponse { success(HOME_DATA_ACTIVE_WITH_MULTIPLE_PSA) },
+        PayinStatusQuery.QUERY_DOCUMENT to apolloResponse { success(PAYIN_STATUS_DATA_NEEDS_SETUP) }
+    )
+
+    @get:Rule
+    val apolloCacheClearRule = ApolloCacheClearRule()
+
+    @Test
+    fun shouldOpenFirstPSALinks() {
```
Comment (MathiasOhrstrom) [inline app/src/androidTest/java/com/hedvig/app/feature/home/ActiveWithMultiplePSAsAndConnectPayment.kt:45]: Seems like we cant open the second link in the same test using Espresso

**Review by hedvigoscar — APPROVED** [review]

**Pull request merged**